### PR TITLE
Postgres sql integration

### PR DESCRIPTION
**Pull request by Tobi-Sheriff**
integrate postgres using sequelize

### CODE REVIEW

File: .env.test
Comment (ajibigad) [inline .env.test]: Change this to db so we are sure db test is passing as well

**Review by ajibigad — CHANGES_REQUESTED** [review]

**Comment by ajibigad** [comment]
I can see merge conflicts. Did you pull the main branch into this branch? You need to your changes are on top of the main branch HEAD

**Comment by Tobi-Sheriff** [comment]
No I didn't pull the main branch, GitHub automation test failed for my initial PR so maybe that's what caused it

File: .env.test
Comment (Tobi-Sheriff) [reply .env.test]: Do you mean I should change the storage from file to dB for the test? Won't that mean that I won't be using the fileRepository again?

And to do that, should I create a new dbTestTRepository or use the one that I am using for dB. Which will require changes in the adapter right?

Comment (ajibigad) [reply .env.test]: The tests should pass without any modifications. We should be able to switch between file and db store type for test and service without any modifications

File: .github/workflows/jest-tests.yml
Comment (ajibigad) [inline .github/workflows/jest-tests.yml:1]: I thought we had committed this change before? I can see the file here https://github.com/Tobi-Sheriff/movie-archive/blob/main/.github/workflows/jest-tests.yml

Comment (ajibigad) [reply .github/workflows/jest-tests.yml:1]: Your branching is probably broken. Can you ensure all changes in the main branch are pulled into your branch first before adding your new changes? This will enable us see the diff properly

File: README.md
Comment (ajibigad) [inline README.md]: Same here. Something is wrong. These changes are already in your main branch

File: .env.test
Comment (Tobi-Sheriff) [reply .env.test]: Oh okay

File: .github/workflows/jest-tests.yml
Comment (Tobi-Sheriff) [reply .github/workflows/jest-tests.yml:1]: > I thought we had committed this change before? I can see the file here https://github.com/Tobi-Sheriff/movie-archive/blob/main/.github/workflows/jest-tests.yml

Yes, i didnt make any changes there, all did was to clean some of the uneccessary comments

Comment (Tobi-Sheriff) [reply .github/workflows/jest-tests.yml:1]: > Your branching is probably broken. Can you ensure all changes in the main branch are pulled into your branch first before adding your new changes? This will enable us see the diff properly

A won't that mean that i will loose all my current work / changes?

File: README.md
Comment (Tobi-Sheriff) [reply README.md]: Oh this one, i think i was trying to do something which i cant recollect now but the only changes in this readme file i think is the second line. Oh i thin i remember, it was when we wanted to test the git automation so i was looking for a file where i can just make random changes and then make a PR to test the git automation test

Comment (ajibigad) [reply README.md]: I know. I mean you did this before and have already merged this change into the main branch. Why is it showing again in this PR?

File: .github/workflows/jest-tests.yml
Comment (ajibigad) [reply .github/workflows/jest-tests.yml:1]: The branch looks broken to me. You can do a soft reset back to the same head as master or the last common head your branch shares with master. Stash your changes, pull main into your branch, apply your stash, fix all merge conflicts. We shouldn't be seeing changes you already have in main in this branch

File: repositories/dbCommentRepository.js
Comment (ajibigad) [inline repositories/dbCommentRepository.js]: Let's remove this if it is no longer required

File: repositories/dbCommentRepository.js
Comment (ajibigad) [inline repositories/dbCommentRepository.js]: Let's not do this, deleting the records should be fine

File: repositories/dbCommentRepository.js
Comment (ajibigad) [inline repositories/dbCommentRepository.js]: We shouldn't do this, the function should work regardless the env. It should just add comments with their movie id already specified

File: index.js
Comment (ajibigad) [inline index.js]: Why async?

File: controllers/movies.js
Comment (ajibigad) [inline controllers/movies.js]: Why?

File: repositories/dbMovieRepository.js
Comment (ajibigad) [inline repositories/dbMovieRepository.js]: Confirm if needed. If not, lets delete it

File: repositories/dbMovieRepository.js
Comment (ajibigad) [inline repositories/dbMovieRepository.js]: Can we remove these?

File: repositories/dbMovieRepository.js
Comment (ajibigad) [inline repositories/dbMovieRepository.js]: No need for this

File: repositories/fileCommentRepository.js
Comment (ajibigad) [inline repositories/fileCommentRepository.js]: the movie id should already be set in the comment

File: repositories/fileMovieRepository.js
Comment (ajibigad) [inline repositories/fileMovieRepository.js]: Can't we use the id of the last movie as the max id?

File: repositories/dbCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbCommentRepository.js]: Done

File: repositories/dbCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbCommentRepository.js]: Done

File: index.js
Comment (Tobi-Sheriff) [reply index.js]: I was calling a function that requiries waiting for response at one time but no longer needed.

File: controllers/movies.js
Comment (Tobi-Sheriff) [reply controllers/movies.js]: I was honing for consistency in the response but since like you said it's just one object being returned then no need to array. So i have removed the quare brackets

File: repositories/dbMovieRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbMovieRepository.js]: Modified, since it helps eliminating repition

File: repositories/dbMovieRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbMovieRepository.js]: Done

File: repositories/fileCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/fileCommentRepository.js]: Sorry the movie id should be comment id.

File: repositories/fileMovieRepository.js
Comment (Tobi-Sheriff) [reply repositories/fileMovieRepository.js]: Yes, updated that

File: repositories/dbCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbCommentRepository.js]: Okay, removed that

File: .github/workflows/jest-tests.yml
Comment (ajibigad) [inline .github/workflows/jest-tests.yml]: Is this duplication deliberate?

File: package.json
Comment (ajibigad) [inline package.json:18]: What is this used for?

File: repositories/dbCommentRepository.js
Comment (ajibigad) [inline repositories/dbCommentRepository.js]: You are still not re-raising these errors after logging. This is dangerous

File: repositories/dbCommentRepository.js
Comment (ajibigad) [inline repositories/dbCommentRepository.js]: Please re-raise

File: repositories/dbCommentRepository.js
Comment (ajibigad) [inline repositories/dbCommentRepository.js]: Please re-raise

File: repositories/dbMovieRepository.js
Comment (ajibigad) [inline repositories/dbMovieRepository.js]: Please re-raise

File: seeds/seedGenerator.js
Comment (ajibigad) [inline seeds/seedGenerator.js]: Why?? Thought we discussed not breaking these boundaries??

File: test-connection.js
Comment (ajibigad) [inline test-connection.js]: Do we need this file?

**Review by ajibigad — COMMENTED** [review]
Please re-raise exceptions in your repositories. Not re-raising them will lead to nigthmares when debugging issues.

File: .github/workflows/jest-tests.yml
Comment (Tobi-Sheriff) [reply .github/workflows/jest-tests.yml]: Oh it's a mistake , must have been during the rebase

File: package.json
Comment (Tobi-Sheriff) [reply package.json:18]: Oh it's a Postgres client but since I'm using sequelize and it's been used in the sequelize instance

File: repositories/dbCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbCommentRepository.js]: I am working on that now

File: repositories/dbCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbCommentRepository.js]: Okay

File: repositories/dbCommentRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbCommentRepository.js]: Okay

File: repositories/dbMovieRepository.js
Comment (Tobi-Sheriff) [reply repositories/dbMovieRepository.js]: Okay

File: test-connection.js
Comment (Tobi-Sheriff) [reply test-connection.js]: Oh not really, i was using it to test connection when i had issue with the db connection

File: seeds/seedGenerator.js
Comment (Tobi-Sheriff) [reply seeds/seedGenerator.js]: Yes, removed

**Review by ajibigad — APPROVED** [review]

**Comment by ajibigad** [comment]
Let's fix the error handling. Remove try/catch in the repositories and controllers.

**Comment by ajibigad** [comment]
Also write tests for the error handling logic. The service should return 500 internal error whenever there is an uncaught exception

**Pull request merged**
